crates/zed/src/main.rs

```diff
@@ -312,7 +312,7 @@ fn main() {
     {
         if env::var("ZED_STATELESS").is_err() {
             if crate::zed::listen_for_cli_connections(open_listener.clone()).is_err() {
-                println!("zed is already running");
+                println!("zed is already running {:?}", crate::zed::listen_for_cli_connections(open_listener.clone()));
```

Comment:
This is slightly wrong; you're calling this function the second time to get the error value, but it's not guaranteed that it'll fail again (even though it most likely will). Plus, `listen_for_cli_connections` has side effects.
I think it'd be better to just store the error and print it:
```
            if let Err(connection_error) = crate::zed::listen_for_cli_connections(open_listener.clone()) {
                println!("zed is already running: {connection_error:?}");
                return;
            }
```